src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -145,21 +147,52 @@ public void deleteEnvironment(@Verify(Environment.class) String envId) {
             throw new NotFoundException(i18n.tr("No such environment: {0}", envId));
         }
 
-        CandlepinQuery<Consumer> consumers = this.envCurator.getEnvironmentConsumers(environment);
+        List<Consumer> consumers = this.envCurator.getEnvironmentConsumers(environment).list();
+        deleteConsumers(environment, consumers);
 
+        log.info("Deleting environment: {}", environment);
+        this.envCurator.delete(environment);
+    }
+
+    private void deleteConsumers(Environment environment, List<Consumer> consumers) {
+        if (consumers.isEmpty()) {
+            log.info("No consumers found in environment: {}", environment);
+            return;
+        }
         // Cleanup all consumers and their entitlements:
         log.info("Deleting consumers in environment {}", environment);
-        for (Consumer c : consumers.list()) {
+        for (Consumer c : consumers) {
             log.info("Deleting consumer: {}", c);
 
             // We're about to delete these consumers; no need to regen/dirty their dependent
             // entitlements or recalculate status.
             this.poolManager.revokeAllEntitlements(c, false);
             this.consumerCurator.delete(c);
         }
+        revokeIdentityCertificates(consumers);
+        revokeContentAccessCertificates(consumers);
```

Comment:
Both of these are only revoking the serials, but what we want is to entirely delete the certificates _and_ revoke the serials

Reply:
Done.